File: frontend/src/types/recommendation.ts
/** Mirrors backend schemas/recommendation.py */

export interface ToolCallTrace {
  tool_name: string
  arguments: Record<string, unknown>
  result: Record<string, unknown>
  execution_ms: number
}

export interface AgentReasoning {
  agent_name: string
  narrative: string
  output_signal: Record<string, unknown>
  tool_calls: ToolCallTrace[]
  confidence_contrib: number | null
  execution_ms: number
}

export type RecommendationStatus =
  | 'pending'
  | 'auto_approved'
  | 'approved'
  | 'rejected'
  | 'escalated'
  | 'applied'

export interface Recommendation {
  id: string
  product_id: string
  product_name: string
  product_sku: string
  current_price: number
  recommended_price: number
  price_change_pct: number
  confidence_score: number
  strategy_label: string
  rationale_summary: string
  status: RecommendationStatus
  created_at: string
}

export interface RecommendationDetail extends Recommendation {
  agent_reasoning: AgentReasoning[]
  reviewed_by: string | null
  reviewed_at: string | null
  review_note: string | null
}

File: backend/app/schemas/recommendation.py
"""
recommendation.py — Request/response shapes for recommendations + approval workflow.

The RecommendationDetailOut includes the full embedded agent reasoning,
which powers the Recommendation Detail page in the frontend.

Connected to: routes/recommendations.py, controllers/recommendation_controller.py
"""
from datetime import datetime
from pydantic import BaseModel


class ToolCallTraceOut(BaseModel):
    tool_name: str
    arguments: dict
    result: dict
    execution_ms: int


class AgentReasoningOut(BaseModel):
    agent_name: str
    narrative: str
    output_signal: dict
    tool_calls: list[ToolCallTraceOut]
    confidence_contrib: float | None
    execution_ms: int


class RecommendationOut(BaseModel):
    """Summary view — shown in the recommendations list."""
    id: str
    product_id: str
    product_name: str               # joined from product
    product_sku: str
    current_price: float
    recommended_price: float
    price_change_pct: float
    confidence_score: float
    strategy_label: str
    rationale_summary: str
    status: str
    created_at: datetime


class RecommendationDetailOut(RecommendationOut):
    """Full detail view — includes all agent reasoning."""
    agent_reasoning: list[AgentReasoningOut]
    reviewed_by: str | None
    reviewed_at: datetime | None
    review_note: str | None


class ApproveRequest(BaseModel):
    note: str = ""


class RejectRequest(BaseModel):
    reason: str                     # required — stored in audit log
    note: str = ""


class ModifyRequest(BaseModel):
    """Analyst overrides the AI price and approves."""
    override_price: float
    note: str = ""
